### ISSUE TITLE
Multiple optimizers in AC-shared network architecture

### ISSUE THREAD

**Issue by SamTov** [SwarmRL/SwarmRL#66]
**What feature would you like to see added?**
In our shared network architecture, there is currently no way to use multiple optimizers in training. This could be important when, for example, we want to train a critic agent with a different learning rate to the actor, as can be required in some problems.

**Solution**
We can take a dictionary of optimisers to the `FlaxModel` class, add these to the train state, and then use many optimizers. In cases where someone actually wants to do this, it means they will need to use some naming convention to assign certain parmeters to a certain optimizer.
